### PR TITLE
addressed some issues

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
should close:
* #2683 
* #2666 

and should address the first part of #2583

File: solutions/usaco-1063.mdx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		for (int c = 0; c < cowNum; c++) {
+			cows[c] = Arrays.stream(read.readLine().split(" "))
+					.mapToInt(Integer::parseInt).toArray();
+
+			seenX.add(cows[c][0]);
+			seenY.add(cows[c][1]);
+		}
+
+		/*
+		 * because all the x's and y's are distinct,
```
Comment (ryanchou-dev) [inline solutions/usaco-1063.mdx:38]:
```suggestion
		 * because all the xs and ys are distinct,
```

File: solutions/usaco-1063.mdx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+title: Rectangular Pasture
+author: Kevin Sheng
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_prob2_silver_dec20.html)
+
+## Video Solution
+
+<Youtube id="AH1wyxq8nPM" />
+
```
Comment (ryanchou-dev) [inline solutions/usaco-1063.mdx:13]: maybe add an implementation header?

File: solutions/usaco-1063.mdx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+			}
+		}
+		// count the rectangles with just 1 cow or no cow at all
+		total += cowNum + 1;
+		
+		System.out.println(total);
+	}
+}
+```
+
```
Comment (ryanchou-dev) [inline solutions/usaco-1063.mdx:92]:
```suggestion
```

**Review by ryanchou-dev — APPROVED** [review]

File: solutions/usaco-1063.mdx (hunk 2 above)
Comment (ryanchou-dev) [reply solutions/usaco-1063.mdx:13]: like separate the video solution thingy and the code with

```
## Implementation
```

**Review by ryanchou-dev — APPROVED** [review]
great!

File: solutions/usaco-1063.mdx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+			cows[c] = Arrays.stream(read.readLine().split(" "))
+					.mapToInt(Integer::parseInt).toArray();
+
+			seenX.add(cows[c][0]);
+			seenY.add(cows[c][1]);
+		}
+
+		/*
+		 * because all the xs and ys are distinct,
+		 * we can just sort of "compress" them into just 1, 2, 3, etc..
```
Comment (maggie-j-liu) [inline solutions/usaco-1063.mdx:43]: coordinate compression is explained [here](https://usaco.guide/silver/sorting-custom?lang=java#coordinate-compression), so you can (optionally) add a link?

**Review by maggie-j-liu — APPROVED** [review]

File: solutions/usaco-1063.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/usaco-1063.mdx:43]: sure i can do that

**Pull request merged**